File: DSA/Umang-Singhal/Day1_ques2.c
#include <stdio.h>

int main() {
    int n, k;
    printf("Enter the number of elements in the array: ");
    scanf("%d", &n);
    
    int arr[n]; 
    printf("Enter the elements of the array: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    
    printf("Enter the integer k: ");
    scanf("%d", &k);
    
    int count = 0;
        for (int i = 0; i < n - 1; i++) {
        for (int j = i + 1; j < n; j++) {
            if ((arr[i] + arr[j]) % k == 0) {
                count++;
            }
        }
    }
    
    printf("Number of valid (i, j) pairs: %d\n", count);
    
    return 0;
}

File: DSA/Ananya_Gambhir/Day1_ques2.java
import java.util.*;
public class Day1_ques1 {
    public static int countDivisiblePairs(int[] arr,int k){
        int count = 0;
        int n = arr.length;
        int[] freq = new int[k];
        
        for (int i = 0; i < n; i++) {
            int remainder = arr[i] % k;
            
            int complement = (k - remainder) % k;
            
            count += freq[complement];
            
            freq[remainder]++;
        }        
        return count;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        System.out.println("Enter the number of elements in the array:");
        int n = sc.nextInt();
        
        int[] arr = new int[n];
        
        System.out.println("Enter the elements of the array:");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        
        System.out.println("Enter the value of k:");
        int k = sc.nextInt();
        
        System.out.println("Number of valid pairs: " + countDivisiblePairs(arr, k));
    }
}
